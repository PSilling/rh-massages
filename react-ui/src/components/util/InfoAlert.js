// react imports
import React, { Component } from 'react';
import PropTypes from 'prop-types';

// util imports
import _t from '../../util/Translations';

/**
 * Bootsrap information alert panel for basic information distribution.
 */
class InfoAlert extends Component {

  render() {
    return (
      <div className="alert alert-info alert-dismissible" style={{ 'marginBottom': '15px' }}>
        <button type="button" className="close" aria-label="Close"
          onClick={this.props.onClose} title={ _t.translate('Close') }>
          <span aria-hidden="true">&times;</span>
        </button>
        {this.props.children}
      </div>
    )
  }
}

InfoAlert.propTypes = {
  /** function to be called on dismiss */
  onClose: PropTypes.func.isRequired
};

export default InfoAlert
