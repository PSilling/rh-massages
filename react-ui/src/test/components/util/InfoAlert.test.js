// react imports
import React from 'react';
import TestRenderer from 'react-test-renderer';

// test imports
import InfoAlert from '../../../components/util/InfoAlert';

test('renders content with correct props', () => {
  const testFunction = jest.fn(),
        testRenderer = TestRenderer.create(<InfoAlert onClose={testFunction} />),
        testInstance = testRenderer.root;

  const button = testInstance.findByType('button'),
        treeJSON = testRenderer.toJSON();

  expect(button.props.onClick).toBe(testFunction);
  expect(treeJSON).toMatchSnapshot();
});
