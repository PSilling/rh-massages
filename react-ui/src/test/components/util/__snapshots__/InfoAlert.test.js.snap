// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content with correct props 1`] = `
<div
  className="alert alert-info alert-dismissible"
  style={
    Object {
      "marginBottom": "15px",
    }
  }
>
  <button
    aria-label="Close"
    className="close"
    onClick={[Function]}
    title="Zavřít"
    type="button"
  >
    <span
      aria-hidden="true"
    >
      ×
    </span>
  </button>
</div>
`;
